data/crac-creation/crac-creator-cim/src/main/java/com/farao_community/farao/data/crac_creation/creator/cim/crac_creator/remedial_action/RemedialActionSeriesCreator.java

```diff
@@ -198,6 +202,23 @@ private void addPstRangeAction(String createdRemedialActionId, String createdRem
                 .withInitialTap(pstHelper.getInitialTap())
                 .withTapToAngleConversionMap(pstHelper.getTapToAngleConversionMap());
 
+        // ---- add groupId if present
+        if (cimCracCreationParameters != null && cimCracCreationParameters.getRangeActionGroups() != null) {
+            String groupId = null;
+            for (RangeActionGroup rangeActionGroup : cimCracCreationParameters.getRangeActionGroups()) {
+                for (String raGroupId : rangeActionGroup.getRangeActionsIds()) {
+                    if (raGroupId.equals(createdRemedialActionId)) {
+                        if (groupId != null) {
+                            cracCreationContext.getCreationReport().warn(String.format("GroupId already defined to %s for PST %s, group %s is ignored (only in PST %s).", groupId, createdRemedialActionId, rangeActionGroup, createdRemedialActionId));
```

Comment:
rangeActionGroup is not a String

Reply:
Indeed! I made the change to TRemedialActionAdder too (where i got this block from)

Reply:
Actually, the id of the group is created by calling rangeActionGroup.toString(), and calling the toString in the String.format() method is redundant.